server/apiv2/handlers/tso_keyspace_group.go

```diff
@@ -122,6 +126,55 @@ func DeleteKeyspaceGroupByID(c *gin.Context) {
 	c.JSON(http.StatusOK, nil)
 }
 
+// SplitKeyspaceGroupByIDParams defines the params for splitting a keyspace group.
+type SplitKeyspaceGroupByIDParams struct {
+	ID        uint32   `json:"id"`
+	NewID     uint32   `json:"new-id"`
+	Keyspaces []uint32 `json:"keyspaces"`
+}
+
+// SplitKeyspaceGroupByID splits keyspace group by ID into a new keyspace group with the given new ID.
+// And the keyspaces in the old keyspace group will be moved to the new keyspace group.
+func SplitKeyspaceGroupByID(c *gin.Context) {
+	splitParams := &SplitKeyspaceGroupByIDParams{}
+	err := c.BindJSON(splitParams)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, errs.ErrBindJSON.Wrap(err).GenWithStackByCause())
+		return
+	}
+	if !isValid(splitParams.ID) || !isValid(splitParams.NewID) {
+		c.AbortWithStatusJSON(http.StatusBadRequest, "invalid keyspace group id")
+		return
+	}
+
+	svr := c.MustGet(middlewares.ServerContextKey).(*server.Server)
+	manager := svr.GetKeyspaceGroupManager()
+	err = manager.SplitKeyspaceGroupByID(splitParams.ID, splitParams.NewID, splitParams.Keyspaces)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, err.Error())
+		return
+	}
+	c.JSON(http.StatusOK, nil)
+}
+
+// FinishSplitKeyspaceByID finishes split keyspace group by ID.
+func FinishSplitKeyspaceByID(c *gin.Context) {
```

Comment:
Who calls FinishSplitKeyspaceByID -- tso.KeyspaceGroupManager or human? For the latter, how do we know the split is successfully completed on the tso.KeyspaceGroupManager side?

Reply:
It will be used by `tso.KeyspaceGroupManager` to finish the split.